hyperspy/io_plugins/nexus.py

```
@@ -0,0 +1,1251 @@
+        
+    return toreturn
+
+def _text_split(s, sep):   
+    """ Split a string based of list of seperators
+    
+    Parameters
+    ----------
+    s   : str 
+    sep : str  - seperator or list of seperators e.g. '.' or ['_','/']
+    
+    Returns
+    -------
+    list 
+       String sections split based on the seperators
+    
+    """
+    stack = [s]
+    for char in sep:
+        pieces = []
+        for substr in stack:
+            pieces.extend(substr.split(char))
+        stack = pieces
+    if '' in stack:
+        stack.remove('')
+    return stack
+
+def _getlink(h5group,rootkey,key):    
+    """ Return the link target path.
+
+    If a hdf group is a soft link or has a target attribute  
+    this method will return the target path. If no link is found
+    return None.
+    
+    
+    Returns
+    -------
+    str
+        Soft link path if it exists, otherwise None
+    
+    """
+    _target=None
+    if rootkey != '/':
+        if isinstance(h5group,h5py.Group):
+            _link = h5group.get(key, getlink=True)
+            if isinstance(_link, h5py.SoftLink):
+                _target = _link.path
+        if 'target' in h5group.attrs.keys():
+            _target = _parse_from_file(h5group.attrs['target'])
+            if not _target.startswith('/'):
+                _target = '/' + _target
+            if _target == rootkey:
+                _target = None
+
+    return _target
+
+
+def _extract_hdf_dataset(group,dataset,lazy=True):    
+    """ Import data from hdf path 
+    
+    Parameters
+    ----------
+    group : hdf group 
+        group from which to load the dataset
+    dataset : str   
+        path to the dataset within the group
+    lazy    : bool {default:True} 
+        If true use lazy opening, if false read into memory
+    
+    Returns
+    -------
+    dask or numpy array
+        
+    """
+    data = group[dataset]
+    if data.dtype.type is np.string_ or data.dtype.type is np.object_:
+        return None
```

Raise an error instead of return None?

---

The none was to skip  over string arrays and only process numeric .I've moved the logic into the _find_data method to only list numeric datasets.